src/sortStudents.ts

```diff
@@ -1,16 +1,76 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  sortOrder: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+  const sortDirection = (sortOrder === 'asc') ? 1 : -1;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+
+      sortedStudents.sort(
+        (firstStudent: Student, secondStudent: Student) => (
+          firstStudent.name.localeCompare(secondStudent.name)
+        ),
+      );
+      break;
+
+    case SortType.Surname:
+      return sortedStudents.sort(
+        (firstStudent: Student, secondStudent: Student) => (
+          firstStudent.surname.localeCompare(secondStudent.surname)
+        ),
+      );
```

Comment:
you can simplify and reduce the code. please fix all possible cases
```suggestion
    case SortType.Name:
    case SortType.Surname:
      return sortedStudents.sort(
        (firstStudent: Student, secondStudent: Student) => (
          firstStudent[sortBy].localeCompare(secondStudent[sortBy])
        ),
      );
      break;
```